File: android/src/main/java/com/kalturaplayerrn/model/SubtitleStyling.java
package com.kaltura.react_native_kplayer.model;

import android.graphics.Color;
import android.text.Layout;

import com.kaltura.playkit.player.SubtitleStyleSettings;

public class SubtitleStyling {
    public String subtitleStyleName;
    public Config config;

    public class Config {
        public String subtitleTextColor;
        public String subtitleBackgroundColor;
        public String subtitleWindowColor;
        public String subtitleEdgeColor;
        public String subtitleTextSizeFraction;
        public String subtitleStyleTypeface;
        public String subtitleEdgeType;
        public Boolean overrideInlineCueConfig;
        public Integer verticalPositionPercentage;
        public Integer horizontalPositionPercentage;
        public String horizontalAlignment;

        public Integer getColorDefaultWhite(String color) {
            if (color != null) {
                return Color.parseColor(color);
            } else {
                return Color.WHITE;
            }
        }

        public Integer getColorDefaultBlack(String color) {
            if (color != null) {
                return Color.parseColor(color);
            } else {
                return Color.BLACK;
            }
        }

        public SubtitleStyleSettings.SubtitleTextSizeFraction getSubtitleTextSizeFraction() {
            if (SubtitleStyleSettings.SubtitleTextSizeFraction.valueOf(subtitleTextSizeFraction) != null) {
                return SubtitleStyleSettings.SubtitleTextSizeFraction.valueOf(subtitleTextSizeFraction);
            }
            return SubtitleStyleSettings.SubtitleTextSizeFraction.SUBTITLE_FRACTION_100;
        }

        public SubtitleStyleSettings.SubtitleStyleTypeface getSubtitleStyleTypeface() {
            if (SubtitleStyleSettings.SubtitleStyleTypeface.valueOf(subtitleStyleTypeface) != null) {
                return SubtitleStyleSettings.SubtitleStyleTypeface.valueOf(subtitleStyleTypeface);
            }
            return SubtitleStyleSettings.SubtitleStyleTypeface.DEFAULT;
        }

        public SubtitleStyleSettings.SubtitleStyleEdgeType getSubtitleEdgeType() {
            if (SubtitleStyleSettings.SubtitleStyleEdgeType.valueOf(subtitleEdgeType) != null) {
                return SubtitleStyleSettings.SubtitleStyleEdgeType.valueOf(subtitleEdgeType);
            }
            return SubtitleStyleSettings.SubtitleStyleEdgeType.EDGE_TYPE_NONE;
        }

        public Layout.Alignment getHorizontalAlignment() {
            if (Layout.Alignment.valueOf(horizontalAlignment) != null) {
                return Layout.Alignment.valueOf(horizontalAlignment);
            }
            return Layout.Alignment.ALIGN_CENTER;
        }
    }
}

File: android/src/main/java/com/reactnativekalturaplayer/model/SubtitleStyling.kt
package com.reactnativekalturaplayer.model

import android.graphics.Color
import android.text.Layout
import android.text.TextUtils
import androidx.annotation.Nullable
import androidx.core.graphics.ColorUtils
import com.kaltura.playkit.player.SubtitleStyleSettings

class SubtitleStyling {
    var subtitleStyleName: String? = null
    var subtitleTextColor: String? = null
    var subtitleBackgroundColor: String? = null
    var subtitleWindowColor: String? = null
    var subtitleEdgeColor: String? = null
    private var subtitleTextSizeFraction: String? = null
    private var subtitleStyleTypeface: String? = null
    private var subtitleEdgeType: String? = null
    var overrideInlineCueConfig: Boolean = false
    var verticalPositionPercentage: Int? = null
    var horizontalPositionPercentage: Int? = null
    private var horizontalAlignment: String? = null

    @Nullable
    fun getStringToColor(color: String?): Int? {
        return if (!TextUtils.isEmpty(color)) {
            Color.parseColor(color)
        } else {
            null
        }
    }

    fun getSubtitleTextSizeFraction(): SubtitleStyleSettings.SubtitleTextSizeFraction {
        subtitleTextSizeFraction?.let { fraction ->
            if (SubtitleStyleSettings.SubtitleTextSizeFraction.values().map { it.name }.contains(fraction)) {
                return SubtitleStyleSettings.SubtitleTextSizeFraction.valueOf(fraction)
            }
        }
        return SubtitleStyleSettings.SubtitleTextSizeFraction.SUBTITLE_FRACTION_100
    }

    fun getSubtitleStyleTypeface(): SubtitleStyleSettings.SubtitleStyleTypeface {
        subtitleStyleTypeface?.let { typeface ->
            if (SubtitleStyleSettings.SubtitleStyleTypeface.values().map { it.name }.contains(typeface)) {
                return SubtitleStyleSettings.SubtitleStyleTypeface.valueOf(typeface)
            }
        }
        return SubtitleStyleSettings.SubtitleStyleTypeface.DEFAULT
    }

    fun getSubtitleEdgeType(): SubtitleStyleSettings.SubtitleStyleEdgeType {
        subtitleEdgeType?.let { edgeType ->
            if (SubtitleStyleSettings.SubtitleStyleTypeface.values().map { it.name }.contains(edgeType)) {
                return SubtitleStyleSettings.SubtitleStyleEdgeType.valueOf(edgeType)
            }
        }
        return SubtitleStyleSettings.SubtitleStyleEdgeType.EDGE_TYPE_NONE
    }

    fun getHorizontalAlignment(): Layout.Alignment {
        horizontalAlignment?.let { alignment ->
            if (Layout.Alignment.values().map { it.name }.contains(alignment)) {
                return Layout.Alignment.valueOf(alignment)
            }
        }
        return Layout.Alignment.ALIGN_CENTER
    }

    fun isDarkColor(color: Int): Boolean {
        return ColorUtils.calculateLuminance(color) < 0.5
    }
}
